protocol/app/upgrades/v5.0.0/upgrade.go

```diff
@@ -98,11 +99,41 @@ func blockRateLimitConfigUpdate(
 	)
 }
 
+func negativeTncSubaccountSeenAtBlockUpgrade(
+	ctx sdk.Context,
+	perpetualsKeeper perptypes.PerpetualsKeeper,
+	subaccountsKeeper satypes.SubaccountsKeeper,
+) {
+	// Get block height stored by v4.x.x.
+	blockHeight, exists := subaccountsKeeper.LegacyGetNegativeTncSubaccountSeenAtBlock(ctx)
+	// If no block height was stored in the legacy store, no migration needed.
+	if !exists {
+		return
+	}
+
+	// If there are no perpetuals, then no new state needs to be stored, as there can be no
+	// negative tnc subaccounts w/o perpetuals.
+	perpetuals := perpetualsKeeper.GetAllPerpetuals(ctx)
+	if len(perpetuals) == 0 {
+		return
+	}
+
+	// Migrate the value from the legacy store to the new store.
+	if err := subaccountsKeeper.SetNegativeTncSubaccountSeenAtBlock(
+		ctx,
+		perpetuals[0].Params.Id, // must be a cross-margined perpetual due to `perpetualsUpgrade`.
```

Comment:
for my understanding, we are just using perp 0 (btc) which should be cross margined? 
Might be nice to add a panic check to assert that perp 0 is cross margined but don't feel too strongly

Reply:
>we are just using perp 0 (btc) which should be cross margined?
Not exactly. Since this upgrade contains both the "add market type" upgrade and the "change negative subaccount seen" upgrade, all perpetuals (as long as one exists) must be cross-margined. So perpetual 0 on any deployment of the chain must be cross-margined at this point.